frontend/packages/dev-console/src/components/edit-application/EditApplicationComponent.tsx

```diff
@@ -26,13 +26,24 @@ const EditApplicationComponent: React.FunctionComponent<EditApplicationComponent
   const { appName, resources } = props;
   const appLabel = resources.editAppResource?.data?.metadata?.labels?.['app.kubernetes.io/name'];
 
-  const getAssociatedResource = (resourcesObj: WatchK8sResultsObject<K8sResourceKind[]>) => {
-    const associatedRes = resourcesObj.data?.find(
-      (ob) =>
-        ob.metadata.name === appName ||
-        ob.metadata.name === appLabel ||
-        ob.metadata.labels['app.kubernetes.io/name'] === appLabel,
+  const filterAssociatedResource = (obj: K8sResourceKind) => {
+    return (
+      obj.metadata.name === appName ||
+      obj.metadata.name === appLabel ||
+      obj.metadata.labels['app.kubernetes.io/name'] === appLabel
```

Comment:
Didn't tested this yet myself, but didn't you want to check the appName or appLabel also against `app.kubernetes.io/instance` ? :thinking: 

And `appLabel` could be undefined, so please check that we don't compare undefined === undefined. This would result in an "random" selection.

And please also ensure that this will not fail when `obj.metadata.labels` is not defined.

```suggestion
    return (
      obj.metadata.name === appName ||
      appLabel && (
        obj.metadata.name === appLabel ||
        obj.metadata.labels.?[NAME_LABEL] === appLabel
        obj.metadata.labels.?[INSTANCE_LABEL] === appLabel  // I'm unsure
      )
```